docs/docs/create-source-plugin.md

```diff
@@ -100,3 +100,56 @@ package provides a set of helper functions to generate Node objects with the
 required fields. This includes automatically generating fields like node IDs
 and the `contentDigest` MD5 hash, keeping your code focused on data gathering,
 not boilerplate.
+
+## Advanced
+
+### Adding relationships between nodes
+
+Gatsby source plugins not only create nodes, they also create relationships between nodes that are exposed to GraphQL queries.
+
+There are two ways of adding node relationships in Gatsby: (1) hierarchical (parent-child) or (2) foreign-key based. 
+
+#### Hierarchical relationships
+
+An example of a hierarchical relationship would be a `transformer` plugin changing a file node's markdown string into a `childMarkdownRemark` node that is attached to that fileNode. The markdown node doesn't make sense outside of the file it belongs to. 
+
+When a parent node is deleted, Gatsby deletes all of the child nodes (and their child nodes, and so on). 
```

Comment:
"Because all children nodes are derived from their parent, when a parent node is deleted or changed,..."